play-json/src/main/scala/com/lightbend/lagom/scaladsl/playjson/PlayJsonSerializer.scala

```diff
@@ -66,16 +67,17 @@ private[lagom] final class PlayJsonSerializer(val system: ExtendedActorSystem, r
   override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = {
     val startTime = if (isDebugEnabled) System.nanoTime else 0L
 
-    val i = manifest.lastIndexOf('#')
-    val fromVersion = if (i == -1) 1 else manifest.substring(i + 1).toInt
-    val manifestClassName = if (i == -1) manifest else manifest.substring(0, i)
+    val (fromVersion: Int, manifestClassName: String) = parseManifest(manifest)
 
     val migration = migrations.get(manifestClassName)
 
     val migratedManifest = migration match {
-      case Some(migration) if migration.currentVersion > fromVersion =>
+      case Some(migration) if (migration.currentVersion > fromVersion) =>
         migration.transformClassName(fromVersion, manifestClassName)
-      case None => manifestClassName
+      case Some(migration) if (migration.currentVersion < fromVersion) =>
+        throw new IllegalStateException(s"Migration version ${migration.currentVersion} is " +
+          s"behind version $fromVersion of deserialized type [$manifestClassName]")
```

Comment:
No it doesn't. Shards can move around from node to node for any number of reasons, for example, it's possible that during an upgrade, a non upgraded node could be restarted, and this could trigger a rebalancing that caused shards to be moved off the upgraded node.  The bigger issue though is read sides, there's no correlation in Lagom between which node each write side entity lives on, and which node the read processor that's consuming events for that entity lives on.